strings/knuth_morris_pratt.cpp

```diff
@@ -18,78 +18,84 @@
 #else
 #include <cstring>
 #endif
+#include <cassert>
```

Comment:
Since I can't merge if I am the last one to push a change could you document this include?

```suggestion
#include <cassert> /// for assert()
```